xapian-letor/ranker/ranker.cc

```diff
@@ -390,11 +393,27 @@ Ranker::rank(Xapian::MSet & mset, const string & model_key, const Xapian::Featur
     mset.sort_by_relevance();
 }
 
+vector<double>
+Ranker::xavier_initialisation(int feature_cnt)
+{
+    // Construct a trivial random generator engine:
+    // 469382313 is a random number for which we are getting the best
+    // performance of letor against standard benchmark datasets.
+    default_random_engine generator(469382313);
+    normal_distribution<double> distribution(0.0, sqrt(2.0 /
+						       (1 + feature_cnt)));
```

Comment:
Wrapping between parameters seems more readable than in the middle of a parameter:

```
    normal_distribution<double> distribution(0.0,
                                             sqrt(2.0 / (1 + feature_cnt)));
```